internal/gemalto/key-secure.go

```diff
@@ -0,0 +1,354 @@
+// Copyright 2020 - MinIO, Inc. All rights reserved.
+// Use of this source code is governed by the AGPLv3
+// license that can be found in the LICENSE file.
+
+// Package gemalto implements a key store that fetches/stores
+// cryptographic keys on a Gemalto KeySecure instance.
+package gemalto
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/minio/kes"
+	xhttp "github.com/minio/kes/internal/http"
+)
+
+// Credentials represents a Gemalto KeySecure
+// refresh token that can be used to obtain a
+// short-lived authentication token.
+//
+// A token is valid within either the default root
+// domain (empty) or a specifc domain - e.g. my-domain.
+type Credentials struct {
+	Token  string        // The KeySecure refresh token
+	Domain string        // The KeySecure domain - similar to a Vault Namespace
```

Comment:
```suggestion
	Domain string        // The KeySecure domain - similar to a Gemalto Namespace
```

Reply:
A Gemalto KeySecure domain is kind of equivalent to a Hashicorp Vault Namespace.
A Vault namespace is an internal isolation layer:
```
     Vault
       │
       ├── NS_1 [ users, policies, keys, ...]
       │
       └── NS_2 [ users, policies, keys ...]
```